### PR TITLE
Adding API addMedicalRecord.

### PR DESCRIPTION
**Pull request by ywnam-stephen**
Added addMedicalRecord method. Unified some variables and corrected typo.

### CODE REVIEW

File: src/main/java/org/cnu/realcoding/controller/DogController.java [hunk 1]
```diff
@@ -25,5 +25,18 @@ public Dog getDogByAllKey(@PathVariable String name,
         return dogManagementService.getDogByAllKey(name, ownerName, ownerPhoneNumber);
     }
 
+    @GetMapping("/dogs/{ownerPhoneNumber")
```
Comment (haeramkeem) [inline src/main/java/org/cnu/realcoding/controller/DogController.java]: }

File: src/main/java/org/cnu/realcoding/service/DogManagementService.java [hunk 2]
```diff
@@ -24,13 +24,23 @@ public void insertDog(Dog dog) {
...
+            return mongo_dog;
         }
     }
-    public Dog getDogByPhoneNum(String PhoneNum){
+    public Dog getDogByPhoneNum(String PhoneNum) {
         return dogRepository.getDogByPhoneNum(PhoneNum);
     }
+
+    public void addMedicalRecord(Dog dog, String medicalRecord) {
+        if(dogRepository.check_Exist(dog.getName(), dog.getOwnerName(), dog.getOwnerPhoneNumber())) {
```
Comment (haeramkeem) [inline src/main/java/org/cnu/realcoding/service/DogManagementService.java]: !

File: src/main/java/org/cnu/realcoding/service/DogManagementService.java [hunk 3]
```diff
@@ -24,13 +24,23 @@ public void insertDog(Dog dog) {
...
-            return mongo_dog;
-        }else {
+        if(dogRepository.check_Exist(name, ownerName, ownerPhoneNumber)) {
             throw new DogNotFoundException();
+        }else {
+            return mongo_dog;
         }
     }
-    public Dog getDogByPhoneNum(String PhoneNum){
+    public Dog getDogByPhoneNum(String PhoneNum) {
```
Comment (haeramkeem) [inline src/main/java/org/cnu/realcoding/service/DogManagementService.java:33]: 폰번호가 없는놈 검거

**Review by haeramkeem — COMMENTED** [review]
이거보세유

File: src/main/java/org/cnu/realcoding/controller/DogController.java (hunk 1 above)
Comment (songyoungwoon) [reply src/main/java/org/cnu/realcoding/controller/DogController.java]: why do you use RequestParam? in line 29

**Comment by ywnam-stephen** [comment]
You satisfied now?

File: src/main/java/org/cnu/realcoding/controller/DogController.java (hunk 1 above)
Comment (ywnam-stephen) [reply src/main/java/org/cnu/realcoding/controller/DogController.java]: I found only RequestParam properly works.

**Pull request merged**